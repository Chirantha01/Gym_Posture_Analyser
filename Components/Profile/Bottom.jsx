import { StyleSheet, Text, View , Dimensions } from "react-native";
import React , { useState } from "react";
import Colors from "../Constants/Colors";
import Sizes from "../Constants/Sizes";
import Card from "../Constants/Card";
import { Entypo } from "@expo/vector-icons";
import { FontAwesome } from "@expo/vector-icons";
import Icons from 'react-native-vector-icons/MaterialCommunityIcons';
import CircularProgress from 'react-native-circular-progress-indicator';

const WIDTH =  Dimensions.get("window").width;

export default function Bottom() {
  const [value, setValue] = useState(85);

  return (
    //<ScrollView contentContainerStyle={{ alignItems: 'center', backgroundColor: "white" }}>
    <View style={styles.bottomContainer}>
      <View style={styles.completeContainer}>
        <Card
          icon={
            <Icons
              name="heart-pulse"
              size={50}
              color="red"
            />
            
          }
          cardTextOne="2"
          cardText="Health"
          style={{ backgroundColor: "#93E9BE" ,marginLeft:WIDTH*0.035 , marginRight:WIDTH*0.01}}
        />
        <Card
          icon={
            <Icons name="trophy" size={50} color="gold" />
          }
          cardTextOne="1"
          cardText="Achievements"
          style={{ backgroundColor: "#FFE4C4" , marginLeft:WIDTH*0.01 , marginRight:WIDTH*0.02}}
        />
      </View>
      <View className="flex-row m-3 rounded-3xl p-3 bg-fuchsia-200">
        <CircularProgress
          radius={60}
          value={value}
          textColor='#222'
          fontSize={20}
          valueSuffix={'%'}
          activeStrokeColor={'tomato'}
          inActiveStrokeOpacity={0.2}
          duration={4000}
        />
        <View className="justify-center mx-5">
          <Text className="font-bold text-2xl mb-2">Weakly Progress</Text>
          <Text className="font-semibold text-xl">Time Spent : 15 hours</Text>
          <Text className="font-semibold text-xl">Challenges : 12/15</Text>
        </View>
      </View>
      <View className="mt-2 flex-row">
        <Card
          icon={
            <Icons
              name="heart-pulse"
              size={50}
              color="red"
            />
            
          }
          cardTextOne="2"
          cardText="Health"
          style={{ backgroundColor: "#93E9BE" ,marginLeft:WIDTH*0.035 , marginRight:WIDTH*0.01}}
        />
        <Card
          icon={
            <Icons name="trophy" size={50} color="gold" />
          }
          cardTextOne="1"
          cardText="Achievements"
          style={{ backgroundColor: "#FFE4C4" , marginLeft:WIDTH*0.01 , marginRight:WIDTH*0.02}}
        />
      </View>
      <View style={styles.completeContainer}>
        <Card
          icon={
            <Icons
              name="heart-pulse"
              size={50}
              color="red"
            />
            
          }
          cardTextOne="2"
          cardText="Health"
          style={{ backgroundColor: "#93E9BE" ,marginLeft:WIDTH*0.035 , marginRight:WIDTH*0.01}}
        />
        <Card
          icon={
            <Icons name="trophy" size={50} color="gold" />
          }
          cardTextOne="1"
          cardText="Achievements"
          style={{ backgroundColor: "#FFE4C4" , marginLeft:WIDTH*0.01 , marginRight:WIDTH*0.02}}
        />
      </View>
      <View style={styles.bottomSection} className="p-3 bg-green-300 rounded-3xl w-8/12 mx-auto">
        <Text style={styles.bottomSectionText}>Start Exercising Now</Text>
      </View>
    </View>
    //</ScrollView>
  );
}

const styles = StyleSheet.create({
  bottomContainer: {
    marginTop: Sizes.medium,
    width:WIDTH,
    flexDirection:"column"
  },
  completeContainer: {
    flexDirection: "row",
    marginTop: Sizes.xs,
  },
  card: {
    backgroundColor: Colors.secondary,
  },
  bottomSection: {
    justifyContent: "center",
    alignItems: "center",
    marginTop: Sizes.medium,
  },
  bottomSectionText: {
    fontWeight: "bold",
    fontSize: Sizes.smedium,
    color: Colors.darkGray,
    marginBottom: 5,
    borderBottomColor: Colors.darkGray,
  }
});