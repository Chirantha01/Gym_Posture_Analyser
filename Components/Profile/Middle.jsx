import { StyleSheet, Text, View, Image } from "react-native";
import React from "react";
import { AntDesign } from "@expo/vector-icons";
import Colors from "../Constants/Colors";
import { Entypo } from "@expo/vector-icons";
import { FontAwesome } from "@expo/vector-icons";

export default function Middle() {
  return (
    <View style={styles.main}>
      <View style={styles.imageContainer}>
        <Image style={styles.image} source={require("../../assets/user.jpg")} />
        <Text style={{ fontSize: 30, color: "white", fontWeight: "bold" }}>
          Sansa Stark
        </Text>
        <Text
          style={{ fontSize: 16, fontWeight: "500" }} className="text-gray-300"
        >
          sansa2001@gmail.com
        </Text>
      </View>

      <View style={styles.middleSectionTextContainer}>
        <View style={styles.middleSectionText}>
          <Text style={styles.toptext} >Age</Text>
          <Text style={styles.bottomtext} className="text-gray-300">23</Text>
        </View>
        <View style={styles.middleSectionText}>
          <Text style={styles.toptext}>Weight</Text>
          <Text style={styles.bottomtext} className="text-gray-300">55Kg</Text>
        </View>
        <View style={styles.middleSectionText}>
          <Text style={styles.toptext}>Height</Text>
          <Text style={styles.bottomtext} className="text-gray-300">5' 6"</Text>
        </View>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  main: {
    marginTop: 30,
  },
  imageContainer: {
    justifyContent: "center",
    alignItems: "center",
  },
  image: {
    width: 150,
    height: 150,
    borderRadius: 50,
    marginBottom: 5,
  },
  middleSectionTextContainer: {
    flexDirection: "row",
    justifyContent: "space-evenly",
    marginTop: 20,
  },
  middleSectionText: {
    justifyContent: "center",
    alignItems: "center",
  },
  toptext: {
    fontSize: 18,
    color:  Colors.white,
    fontWeight: "bold",
  },
  bottomtext: {
    fontSize: 16,
    fontWeight: "700",
  },
});
