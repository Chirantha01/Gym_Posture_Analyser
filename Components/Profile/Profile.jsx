import { StyleSheet, Text, View, Image, ScrollView, Pressable, Animated , Alert} from "react-native";
import React, { useRef } from "react";
import ImagePlaceholder from '../../assets/image-placeholder.jpg';

export default function Profile({onLogOut}) {
  const name = "John Doe";
  const email = "johnDoe@email.com";
  const birthDay = "01/01/2001";
  const age = 20;
  const height = 180;
  const weight = 70;


  // Refs for each setting animation
  const editProfileScale = useRef(new Animated.Value(1)).current;
  const changePasswordScale = useRef(new Animated.Value(1)).current;
  const logoutScale = useRef(new Animated.Value(1)).current;

  // Function to handle animations
  const animatePressIn = (scaleValue) => {
    Animated.spring(scaleValue, {
      toValue: 0.95, // Shrink to 95% of original size
      useNativeDriver: true,
    }).start();
  };

  const animatePressOut = (scaleValue) => {
    Animated.spring(scaleValue, {
      toValue: 1, // Return to original size
      friction: 3, // Control the bounce effect
      useNativeDriver: true,
    }).start();
  };

  
  const handleLogoutPress = () => {
    Alert.alert(
      "Confirm Logout",
      "Are you sure you want to log out?",
      [
        {
          text: "Cancel",
          onPress: () => console.log("Logout Cancelled"),
          style: "cancel",
        },
        {
          text: "Yes, Log Out",
          onPress: () => {onLogOut()}, // This will trigger your logout logic
        },
      ],
      { cancelable: true }
    );
  };

  return (
    <ScrollView style={styles.container}>
      <View style={styles.topContainer}>
        <View style={styles.header}>
          <Text style={styles.headerTitle}>My Profile</Text>
        </View>
        <View style={styles.imageContainer}>
          <Image source={ImagePlaceholder} style={styles.profileImage} />
        </View>
        <View>
          <Text style={styles.name}>{name}</Text>
          <Text style={styles.email}>{email}</Text>
          <Text style={styles.birthdayTitle}>
            Birthday: <Text style={styles.birthdayText}>{birthDay}</Text>
          </Text>
        </View>
      </View>
      <View style={styles.infoContainer}>
        <View style={styles.infoBlock}>
          <Text style={styles.infoLabel}>Weight</Text>
          <Text style={styles.infoText}>{weight} kg</Text>
        </View>

        {/* Vertical line */}
        <View style={styles.verticalLine} />
        
        <View style={styles.infoBlock}>
          <Text style={styles.infoLabel}>Age</Text>
          <Text style={styles.infoText}>{age}</Text>
        </View>
        
        {/* Vertical line */}
        <View style={styles.verticalLine} />
        
        <View style={styles.infoBlock}>
          <Text style={styles.infoLabel}>Height</Text>
          <Text style={styles.infoText}>{height} cm</Text>
        </View>
      </View>

      <View style={styles.bottomContainer}>
        {/* Pressable Setting Options with animation */}
        <AnimatedPressable
          scaleValue={editProfileScale}
          onPress={() => console.log("Edit Profile Pressed")}
          iconSource={require("../../assets/profileIcons/Component_39.png")}
          text="Edit my Profile"
        />

        <AnimatedPressable
          scaleValue={changePasswordScale}
          onPress={() => console.log("Change Password Pressed")}
          iconSource={require("../../assets/profileIcons/Component_41.png")}
          text="Change my Password"
        />

        <AnimatedPressable
          scaleValue={logoutScale}
          onPress={handleLogoutPress}
          iconSource={require("../../assets/profileIcons/Component_44.png")}
          text="Logout"
        />
      </View>
    </ScrollView>
  );
}

// Custom component for Animated Pressable
const AnimatedPressable = ({ scaleValue, onPress, iconSource, text }) => {
  return (
    <Animated.View style={{ transform: [{ scale: scaleValue }] }}>
      <Pressable
        onPressIn={() => animatePressIn(scaleValue)}
        onPressOut={() => animatePressOut(scaleValue)}
        onPress={onPress}
        style={styles.settings}
      >
        <Image source={iconSource} style={styles.profileIcon} />
        <Text style={styles.settingText}>{text}</Text>
      </Pressable>
    </Animated.View>
  );
};

// Functions for animation (Move these outside of the Profile function)
const animatePressIn = (scaleValue) => {
  Animated.spring(scaleValue, {
    toValue: 0.95,
    useNativeDriver: true,
  }).start();
};

const animatePressOut = (scaleValue) => {
  Animated.spring(scaleValue, {
    toValue: 1,
    friction: 3,
    useNativeDriver: true,
  }).start();
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: "#232323",
  },
  topContainer: {
    backgroundColor: "#B3A0ff",
    height: 320,
    flex: 1,
  },
  header: {
    width: "100%",
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
    paddingHorizontal: 20,
    paddingBottom: 20,
    paddingTop: 50,
    position: "relative",
  },
  headerTitle: {
    fontSize: 25,
    fontWeight: "bold",
    textAlign: "left",
    color: "white",
    flex: 1,
  },
  imageContainer: {
    justifyContent: "center",
    alignItems: "center",
    marginTop: -20,
  },
  profileImage: {
    width: 125,
    height: 125,
    borderRadius: 100,
  },
  name: {
    fontSize: 25,
    fontWeight: "bold",
    textAlign: "center",
    color: "white",
  },
  email: {
    fontSize: 15,
    textAlign: "center",
    color: "white",
  },
  birthdayTitle: {
    fontSize: 15,
    textAlign: "center",
    color: "white",
    fontWeight: "bold",
  },
  birthdayText: {
    fontSize: 15,
    textAlign: "center",
    color: "white",
    fontWeight: "normal",
  },
  infoContainer: {
    marginTop: -30,
    backgroundColor: "#896CFE",
    height: 60,
    width: "60%",
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "center",
    alignSelf: "center",
    paddingHorizontal: 10,
    borderRadius: 10,
  },
  infoBlock: {
    flexDirection: "column",  // Stack label and value vertically
    alignItems: "center",  // Center the label and value horizontally
    paddingHorizontal: 20,
  },
  infoLabel: {
    color: "white",
    fontSize: 14,
    fontWeight: "bold",
  },
  infoText: {
    color: "white",
    fontSize: 12,
    marginTop: 4, // Space between the label and number
  },
  verticalLine: {
    height: "70%",  // Adjust the height to suit your design
    width: 1,
    backgroundColor: "white",  // Line color
  },
  bottomContainer: {
    marginTop: 20,
  },
  settings: {
    flexDirection: "row",
    paddingLeft: 20,
    paddingTop: 20,
    alignItems: "center",
  },
  profileIcon: {
    width: 30,
    height: 30,
  },
  settingText: {
    paddingTop: 7,
    fontSize: 18,
    fontWeight: "400",
    color: "white",
    paddingLeft: 20,
  },
});
