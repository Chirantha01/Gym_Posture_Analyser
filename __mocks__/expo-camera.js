// __mocks__/expo-camera.js
import React from 'react';

export const Camera = jest.fn(({ children }) => <>{children}</>);
export const requestPermissionsAsync = jest.fn().mockResolvedValue({ status: 'granted' });