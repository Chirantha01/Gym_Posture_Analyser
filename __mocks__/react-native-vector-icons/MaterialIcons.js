import React from 'react';
import { Text } from 'react-native';

const MaterialIcons = (props) => {
    return <Text>{props.name}</Text>;
};

export default MaterialIcons;