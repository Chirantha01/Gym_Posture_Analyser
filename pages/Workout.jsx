import React from 'react';
import { View, Text, Image, StyleSheet, FlatList , StatusBar } from 'react-native';
import { ScrollView } from 'react-native-gesture-handler';

const Workout = () => {
  const data = [
    {
      id: '1',
      image: require('../assets/img-bicep.jpg'),
      title: 'Bicep Curls',
      subtitle: 'Arm Exercise'
    },
    {
      id: '2',
      image: require('../assets/img-squat.jpg'),
      title: 'Squats',
      subtitle: 'Leg Exercises'
    },
    {
      id: '3',
      image: require('../assets/img-sideLateral.jpg'),
      title: 'Side Lateral',
      subtitle: 'Shoulder Exercises'
    },
    {
      id: '4',
      image: require('../assets/img-DumbellPress.jpg'),
      title: 'Dumbell Press',
      subtitle: 'Chest Exercises'
    },
    {
      id: '5',
      image: require('../assets/img-plank.jpg'),
      title: 'Plank',
      subtitle: 'Abdominal Exercises'
    },
    {
      id: '6',
      image: require('../assets/img-SumoSquat.jpg'),
      title: 'Sumo Squat',
      subtitle: 'Leg Exercises'
    },
    {
      id: '7',
      image: require('../assets/img-BenchPress.jpg'),
      title: 'Bench Press',
      subtitle: 'Chest Exercises'
    },
    {
      id: '8',
      image: require('../assets/img-BarbelCurl.jpg'),
      title: 'Barbel Curl',
      subtitle: 'Arm Exercises'
    },
  ];


  const renderItem = ({ item, index }) => {
    return (
      <View style={styles.itemContainer}>
        <Image source={item.image } style={styles.image} />
        <View style={styles.textContainer}>
          <Text style={styles.title}>{item.title}</Text>
          <Text style={styles.subtitle}>{item.subtitle}</Text>
        </View>
      </View>
    );
  };

  return (
    <View>
      <StatusBar style="auto" />
      <FlatList
        data={data}
        contentContainerStyle={styles.container}
        renderItem={renderItem}
        keyExtractor={(item) => item.id}
        numColumns={2}
        ListHeaderComponent={() => (
          <Text className="text-3xl font-bold p-6 justify-center bg-white">Workouts</Text>
        )}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor:'#fff'
  },
  itemContainer: {
    width: '48%',
    height: 300,
    margin: '1%',
    backgroundColor: '#f9f9f9',
    borderRadius: 8,
    overflow: 'hidden',
    padding: 16,
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection:  'column' 
  },
  image: {
    width: '100%',
    height: 200,
    resizeMode: 'cover',
    borderRadius: 8,
  },
  textContainer: {
    padding: 16,
  },
  title: {
    fontWeight: 'bold',
    fontSize: 18,
  },
  subtitle: {
    fontSize: 14,
    color: 'gray',
  },
});

export default Workout;
