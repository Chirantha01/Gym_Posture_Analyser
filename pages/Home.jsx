import React, { useEffect , useState , useCallback } from "react";
import { View, Text, SafeAreaView, StyleSheet, Dimensions, ScrollView, TouchableOpacity, ImageBackground } from "react-native";
import Carousel from '../Components/Image-Carousel/ImageCarouselScreen';
import { useNavigation , useFocusEffect } from '@react-navigation/native';
import { LinearGradient } from 'expo-linear-gradient';
import Workout from './Workout'
import axios from "axios";
import AsyncStorage from '@react-native-async-storage/async-storage';

export default function Home() {
  const [username , setUsername] = useState("")
  console.log("Home screen loaded...");

  const ScreenWidth = Dimensions.get('window').width;

  const navigation = useNavigation();
  const handleArticlePress = (articleName) => {
    navigation.navigate(articleName);
  };

  useEffect(()=>{
    console.log(username);
  }, [username])

  const fetchHome = async(token) =>{
    const res = await axios.get("http://192.168.1.148:4000/home",{headers:{'authorization': `Bearer ${token}`}});
    const data = res.data;
    console.log(data);
    if (data.userName){
      return data.userName;
    }
    else{
      return null;
    }
  }

  useFocusEffect(
    useCallback(() => {
      const fetchUsername = async () => {
        try {
          const token = await AsyncStorage.getItem("jwtToken");
          if (token) {
            const user = await fetchHome(token);
            if (user) {
              setUsername(user);  // Set the username after fetching
            }
          } else {
            console.log("Token not found.");
          }
        } catch (error) {
          console.log("Error fetching Token", error);
        }
      };

      fetchUsername();  // Fetch username when the screen is focused
    }, []) // Empty dependency array to ensure this runs every time the screen is focused
  );

  return (
    <ScrollView style={styles.container}>
      <SafeAreaView>
        <Text style={styles.salutation}>Hi, {username}</Text>
        <Text style={styles.sub_salutation}>It's time to challenge your limits</Text>
        <View>
          <Carousel />

        </View>

        {/* Cube-like Containers for Articles */}
        <View style={styles.articleContainer}>
          <TouchableOpacity style={styles.cube} onPress={() => handleArticlePress('Workout')}>
            <ImageBackground
              source={require('../assets/home_screen/thumb_1.jpg')}  // Replace with your image URL
              style={styles.imageBackground}
              imageStyle={{ borderRadius: 10 }}
            >
              <LinearGradient
                colors={['rgba(0,0,0,0)', 'rgba(0,0,0,1)']}
                style={styles.gradient}
              />
              <Text style={styles.cubeText}>Start my Workout</Text>
              <Text style={styles.cubeDiscription}>Workout with posture detection for a better posture</Text>
            </ImageBackground>
          </TouchableOpacity>

          <TouchableOpacity style={styles.cube} onPress={() => handleArticlePress('Article 2')}>
            <ImageBackground
              source={require('../assets/home_screen/thumb_2.jpg')}  // Replace with your image URL
              style={styles.imageBackground}
              imageStyle={{ borderRadius: 10 }}
            >
              <LinearGradient
                colors={['rgba(0,0,0,0)', 'rgba(0,0,0,1)']}
                style={styles.gradient}
              />
              <Text style={styles.cubeText}>Check my Stats</Text>
              <Text style={styles.cubeDiscription}>Comprehensive summerization of your progress</Text>
            </ImageBackground>
          </TouchableOpacity>

          <TouchableOpacity style={styles.cube} onPress={() => handleArticlePress('Article 2')}>
            <ImageBackground
              source={require('../assets/home_screen/thumb_3.jpg')}  // Replace with your image URL
              style={styles.imageBackground}
              imageStyle={{ borderRadius: 10 }}
            >
              <LinearGradient
                colors={['rgba(0,0,0,0)', 'rgba(0,0,0,1)']}
                style={styles.gradient}
              />
              <Text style={styles.cubeText}>Check my Profile</Text>
              <Text style={styles.cubeDiscription}>Edit your profile details</Text>
            </ImageBackground>
          </TouchableOpacity>

          {/* Add more cubes as needed */}
        </View>
      </SafeAreaView>
    </ScrollView>
  );
}

const styles = StyleSheet.create({
  container: {
    paddingTop: 50,
    backgroundColor: "#232323",
    flex: 1,
  },
  salutation: {
    color: "#896CFE",
    fontSize: 26,
    fontWeight: "bold",
    marginBottom: 0,
    alignSelf: "flex-start",
    marginLeft: 20,
  },
  sub_salutation: {
    color: "white",
    fontSize: 15,
    fontWeight: "semi-bold",
    marginBottom: 20,
    alignSelf: "flex-start",
    marginLeft: 20,
  },
  articleContainer: {
    flexDirection: 'column',  // Display items vertically
    alignItems: 'center',  // Center align items
    marginTop: 20,
  },
  cube: {
    width: '90%',  // Take up 90% of the screen width
    height: Dimensions.get('window').width * 0.4,  // Aspect ratio to keep it like a 'cube'
    marginBottom: 20,
    borderRadius: 10,
    overflow: 'hidden',  // To ensure content stays within cube bounds
  },
  imageBackground: {
    flex: 1,
    justifyContent: 'flex-end',  // Aligns the title to the bottom
  },
  gradient: {
    position: 'absolute',
    bottom: -5,
    left: 0,
    right: 0,
    height: '80%',  // Adjust height for the gradient to cover the bottom half
  },
  cubeText: {
    color: "#E2F163",
    fontSize: 20,
    fontWeight: '500',
    marginLeft: 20,  // Adjust positioning if needed
    marginBottom: 0,
  },
  cubeDiscription : {
    color: "white",
    fontSize: 15,
    fontWeight: '400',
    marginLeft: 20,  // Adjust positioning if needed
    marginRight: 20,
    marginBottom: 15,
  },
});
