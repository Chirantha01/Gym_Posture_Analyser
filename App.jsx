import React, {useState,useEffect} from 'react';
import { View, Text, Dimensions, ActivityIndicator, StyleSheet } from 'react-native';
import HomeCard from './Components/Cards';
import Plank_Model from './pages/Models/Model_plank';
import Bicep_Model from './pages/Models/Model_bicep_curl';
import Squat_Model from './pages/Models/Model_squat';
import PushUp_Model from './pages/Models/Model_push_up';
import LatPullDown_Model from './pages/Models/Model_lat_pull_down';
import Profile from "./Components/Profile/Profile"
import Home from './pages/Home';
import Workout from './pages/Workout'
import { NavigationContainer } from '@react-navigation/native';
import AsyncStorage from '@react-native-async-storage/async-storage';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import { createStackNavigator } from '@react-navigation/stack';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import LoadingScreen from './pages/loading';
import Carousel from './Components/Image-Carousel/ImageCarouselScreen';
import Graph from './pages/Graph';
import Progress from './pages/WorkoutHistory';
import LoginForm from './pages/SignIn';
import SignUpScreen from './pages/SignUp';
import ConfirmOtp from './pages/ConfirmOtp';
import ForgotPassword from './pages/ForgotPassword';
import axios from "axios";

const Stack = createStackNavigator();
const Tab = createBottomTabNavigator();

const PoseApp = () => {
  const [isLoading, setIsLoading] = useState(true);
  const [isAuthenticated , setIsAuthenticated] = useState(false);
  const [isNewUser , setIsNewUser] = useState(false);

  useEffect(() =>{
    const timer = setTimeout(() => {
      setIsLoading(false);
      AsyncStorage.getItem("jwtToken")
        .then(async (token) => {
          if (token){
            
            console.log("Token found" , token);
            tokenStatus = await checkTokenValidity(token);
            console.log("tokenStatus :  ", tokenStatus)
            if (tokenStatus === true){
              console.log("Already authenticated")
              setIsAuthenticated(true);
              console.log("Token Accepted");
            }
            else{
              console.log("Token Declined...");
            }
          }
          else{
            console.log("Token not Found.");
          }
        })
        .catch(error =>{
          console.log("Error fetching Token" , error);
        });
      
    }, 10000);

    return () => clearTimeout(timer);
  }, []);

  const checkTokenValidity = async (token) =>{
    console.log("sending a post request to validate the token");
    const res = await axios.post("http://192.168.1.148:4000/startup",{},{headers:{'authorization': `Bearer ${token}`}});
    console.log("resonse came for the post request");
    console.log("response: " , res.data);
    if (res.data.token_Status === true){
      console.log("token success : True");
      return true;
    }
    else{
      console.log("token success : False");
      return false;
    }
  }

  const onSignIn =() =>{
    setIsAuthenticated(true);
  }

  const onSignUp = () =>{
    setIsAuthenticated(true);
  }

  const onSwitchToSignUp = () =>{
    setIsNewUser(true);
  }

  const onSwitchToSignIn = () =>{
    setIsNewUser(false);
  }

  const onLogOut = () =>{
    console.log("inside onLogOut funtion")
    setIsAuthenticated(false);
    setIsNewUser(false);
  }

  if (isLoading) {
    return <LoadingScreen />;
  }

  if (!isAuthenticated) {
    return (
      <NavigationContainer>
        <Stack.Navigator screenOptions={{ headerShown: false }}>
          {isNewUser ? (
            <Stack.Screen
              name="SignUpScreen"
              component={() => <SignUpScreen onSignUp={onSignUp} onSwitchToSignIn={onSwitchToSignIn} />}
            />
          ) : (
            <Stack.Screen name="LoginForm">
              {() => <LoginForm onSignIn={onSignIn} onSwitchToSignUp={onSwitchToSignUp} />}
            </Stack.Screen>
          )}
          <Stack.Screen
            name="ForgotPassword"
            component={ForgotPassword}
          />
          <Stack.Screen
            name="ConfirmOtp"
            component={ConfirmOtp}
          />
        </Stack.Navigator>
      </NavigationContainer>
    );
  }

return (
  <NavigationContainer>
      <Stack.Navigator
        screenOptions={{
          headerShown: false
        }}
      >

        <Stack.Screen
          name="Main"
          children={() => <TabNavigator onLogOut={onLogOut} />}
          options={{hearderShown : false }} // Main screen title
        />
        <Stack.Screen
          name="LogInForm"
          component={LoginForm}
          options={{hearderShown : false }}
        />
        <Stack.Screen
          name="ForgotPassword"
          component={ForgotPassword}
          options={{hearderShown : false }} // Main screen title
        />
        <Stack.Screen
          name="ConfirmOtp"
          component={ConfirmOtp}
        />
        <Stack.Screen
          name="Model_plank"
          component={Plank_Model} // Plank model component
          options={{ headerTitle: 'Plank_Model', headerTitleAlign: 'center' }}
        />
        <Stack.Screen
          name="Model_bicep"
          component={Bicep_Model} // Bicep Curl model component
          options={{ headerTitle: 'Bicep_Model', headerTitleAlign: 'center' }}
        />
        <Stack.Screen
          name="Model_squat"
          component={Squat_Model} // Squat model component
          options={{ headerTitle: 'Squat_Model', headerTitleAlign: 'center' }}
        />
        <Stack.Screen
          name="Model_push_up"
          component={PushUp_Model} // Pushup model component
          options={{ headerTitle: 'Push_Up_Model', headerTitleAlign: 'center' }}
        />
        <Stack.Screen
          name="Model_lat_pull_down"
          component={LatPullDown_Model}
          options={{ headerTitle: 'Lat_Pull_Down_Model', headerTitleAlign: 'center' }}
        />
        <Stack.Screen
          name="Workout"
          component={Workout} // Lat pull down model component
          // options={{ headerTitle: 'Lat_Pull_Down_Model', headerTitleAlign: 'center' }}
        />
      </Stack.Navigator>
    </NavigationContainer>
);
};

const TabNavigator = ({ onLogOut }) => {
return (
  <Tab.Navigator
    screenOptions={({ route }) => ({
      tabBarIcon: ({ focused, color, size }) => {
        let iconName;

        if (route.name === 'Home') {
          iconName = focused ? 'home' : 'home';
        } else if (route.name === 'Profile') {
          iconName = focused ? 'person' : 'person-outline';
        } else if (route.name === 'Workout') {
          iconName = focused ? 'sports-mma' : 'sports-mma';
        } else if (route.name === 'Progress') {
          iconName = focused ? 'history' : 'history';
        }

        return <MaterialIcons name={iconName} size={focused ? 35 : 30} color={color} />;
      },
      tabBarActiveTintColor: '#E2F163',
      tabBarInactiveTintColor: 'grey',
      headerShown: false,
      tabBarLabel: () => null,
      tabBarStyle: {
        backgroundColor: '#1A1A1A',
        borderTopWidth: 0,
      },
    })}
  >
    <Tab.Screen name="Home" component={Home} />
    <Tab.Screen name="Workout" component={Workout} />
    <Tab.Screen name="Progress" component={Progress} />
    <Tab.Screen name="Profile" children={() => <Profile onLogOut={onLogOut}/>}/>
  </Tab.Navigator>
);
};


export default PoseApp;
