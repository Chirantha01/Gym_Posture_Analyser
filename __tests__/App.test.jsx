import React, {useState,useEffect} from 'react';
import Plank_Model from './../pages/Models/Model_plank';
import Bicep_Model from './../pages/Models/Model_bicep_curl';
import Squat_Model from './../pages/Models/Model_squat';
import PushUp_Model from './../pages/Models/Model_push_up';
import LatPullDown_Model from './../pages/Models/Model_lat_pull_down';
import Profile from "./../Components/Profile/Profile"
import Home from './../pages/Home';
import Workout from './../pages/Workout'
import { NavigationContainer } from '@react-navigation/native';
import AsyncStorage from '@react-native-async-storage/async-storage';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import { createStackNavigator } from '@react-navigation/stack';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import LoadingScreen from './../pages/loading';
import Progress from './../pages/WorkoutHistory';
import LogInForm from './../pages/SignIn';
import SignUpScreen from './../pages/SignUp';
import axios from "axios";
import { Camera } from 'expo-camera';
import { render } from '@testing-library/react-native';
import App from '../App';

test('renders correctly', () => {
  const { toJSON } = render(<App />);
  expect(toJSON()).toMatchSnapshot();
});