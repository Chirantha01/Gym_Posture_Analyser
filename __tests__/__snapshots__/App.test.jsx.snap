// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../assets/Loading_Screen/loading.jpg",
      }
    }
    style={
      {
        "height": "100%",
        "position": "absolute",
        "resizeMode": "cover",
        "width": "100%",
      }
    }
  />
  <ActivityIndicator
    color="#0000ff"
    size="large"
    style={
      {
        "position": "absolute",
        "top": "50%",
      }
    }
  />
</View>
`;
